### DISCUSSION TITLE
How to enable outline to show access specifier (i.e. public/private/protected)

### DISCUSSION THREAD

**Discussion by YalandHong** [microsoft/vscode-cpptools#7112] [Q&A] [answered]
I tried my best Googling but I only found microsoft/vscode#49995 and microsoft/vscode#23927 discussing this problem. Now they have both been closed, but the outline still show nothing difference between "private" and "public" members.

![Screenshot from 2021-03-05 21-02-02](https://user-images.githubusercontent.com/52960068/110119704-23720080-7df7-11eb-97db-ec95f702a430.png)

Is there any setting for enabling outline to show the `private` and `public` specifiers? Or is this feature not able to be implemented?

**Comment by sean-mcmanus** [comment, marked as answer]
There is no setting. We wanted VS to provide an icon modifier, but they did not. Theoretically, we could add extra text. You could file a feature request.

**Reply by YalandHong** [reply]
Thanks! I opened an issue about this feature request. #7120

**Reply by JoonaV556** [reply]
> There is no setting. We wanted VS to provide an icon modifier, but they did not. Theoretically, we could add extra text. You could file a feature request.

I mainly program in C# and would really like this feature too. It's a shame it is only in the cpptools package.

**Comment by joshwand** [comment]
Upstream dependency is https://github.com/microsoft/language-server-protocol/pull/2003
